packages/server-wallet/src/protocols/__test__/ledger-manager.test.ts

```diff
@@ -62,660 +32,735 @@ beforeEach(async () => {
+          ],
+        },
+      })
+    );
+    it(
+      `marks inconsistent states`,
+      testLedgerCrank({
+        as: 'follower',
+        before: {
+          agreed: {turn: 5, bals: {a: 10, b: 10, c: 2, d: 1}},
+          requests: [
+            ['defund', 'd', 1, 1, 'queued'], // amount doesn't match with d's total
+            ['fund', 'c', 2, 1, 'queued'], // c is funded, but with a different amt
+          ],
+        },
+        after: {
+          agreed: {turn: 5, bals: {a: 10, b: 10, c: 2, d: 1}},
+          requests: [
+            ['defund', 'd', 1, 1, 'inconsistent', {missedOps: 0, lastSeen: 5}],
+            ['fund', 'c', 2, 1, 'inconsistent', {missedOps: 0, lastSeen: 5}],
+          ],
+        },
+      })
+    );
+    it(
+      `will cancel requests that haven't been included`,
+      testLedgerCrank({
+        as: 'follower',
+        before: {
+          agreed: {turn: 5, bals: {a: 10, b: 10}},
+          requests: [
+            ['fund', 'c', 2, 1, 'queued'],
+            ['defund', 'c', 2, 1, 'queued'],
+          ],
+        },
+        after: {
+          agreed: {turn: 5, bals: {a: 10, b: 10}},
+          requests: [
+            ['fund', 'c', 2, 1, 'cancelled', {missedOps: 0, lastSeen: 5}],
+            ['defund', 'c', 2, 1, 'cancelled', {missedOps: 0, lastSeen: 5}],
+          ],
+        },
+      })
+    );
+    it(
+      `won't cancel requests that have been included`,
+      testLedgerCrank({
+        as: 'follower',
+        before: {
+          agreed: {turn: 5, bals: {a: 10, b: 10, c: 3}},
+          requests: [
+            ['fund', 'c', 2, 1, 'pending'], // has been included
+            ['defund', 'c', 2, 1, 'queued'],
+          ],
+        },
+        after: {
+          agreed: {turn: 5, bals: {a: 10, b: 10, c: 3}},
+          requests: [
+            ['fund', 'c', 2, 1, 'succeeded', {missedOps: 0, lastSeen: 5}],
+            ['defund', 'c', 2, 1, 'queued', {missedOps: 0, lastSeen: 5}],
+          ],
+        },
+      })
+    );
+    // TODO: it will mark a mismatched cancellation as inconsistent
+  });
+  describe('in the proposal state', () => {
+    it(
+      `accepts the state if complete overlap`,
+      testLedgerCrank({
+        as: 'follower',
+        before: {
+          agreed: {turn: 5, bals: {a: 10, b: 10, c: 3}},
+          proposed: {turn: 6, bals: {a: 8, b: 10, d: 5}},
+          requests: [
+            ['fund', 'd', 4, 1, 'queued'], // included in proposal
+            ['defund', 'c', 2, 1, 'queued'], // included in proposal
+            ['fund', 'e', 1, 1, 'queued'],
+          ],
+        },
+        after: {
+          agreed: {turn: 6, bals: {a: 8, b: 10, d: 5}},
+          requests: [
+            ['fund', 'd', 4, 1, 'succeeded', {missedOps: 0, lastSeen: 6}],
+            ['defund', 'c', 2, 1, 'succeeded', {missedOps: 0, lastSeen: 6}],
+            ['fund', 'e', 1, 1, 'queued', {missedOps: 1, lastSeen: 6}],
```

Comment:
Is it true that the follower has no way of relaying this request to the leader? If I understand correctly, the leader must send a state to the follower that includes this request.

Reply:
Yes that's true. The ledger algorithm starts from a position where both participants have a queue of requests, that they want to apply to the channel. It's the objective's job to ensure that both participants have the same requests in their queue.

Reply:
If the followers wallet notices that a request is getting a lot of missedOps, it can use that as a trigger to sync/re-send that objective.